src/Features/CSharp/Portable/CodeRefactorings/PullMemberUp/CSharpPullMemberUpCodeRefactoringProvider.cs

```diff
@@ -27,11 +27,14 @@ public CSharpPullMemberUpCodeRefactoringProvider(IPullMemberUpOptionsService ser
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public CSharpPullMemberUpCodeRefactoringProvider() : base(null)
+        public CSharpPullMemberUpCodeRefactoringProvider() : this(service: null)
         {
         }
 
-        protected override async Task<SyntaxNode> GetSelectedNodeAsync(CodeRefactoringContext context)
+        protected override Task<SyntaxNode> GetSelectedNodeAsync(CodeRefactoringContext context)
+            => GetSelectedDeclarationOrVariableAsync(context);
+
+        internal static async Task<SyntaxNode> GetSelectedDeclarationOrVariableAsync(CodeRefactoringContext context)
```

Comment:
def don't lkike this.

Reply:
Just needs to be moved to a common helper?

Reply:
yes.  though:

1. the helper class/member can be in PullMemberUp, if the new feature moves to that namespace.
2. the helper class/member shoudl be in a shared location if we're going to have tehse be seperate features in different namespaces.

Effectively, what i want us to avoid is having specific features call into otehr specific features.  that's a bit of a ugly smell that indicates strange coupling that i would like to not have happen.

Reply:
Ok, added a separate helper that the two share.